public/app/percona/settings/components/Platform/Platform.tsx

```diff
@@ -1,22 +1,83 @@
-import React, { FC } from 'react';
+import React, { FC, useEffect, useMemo, useState } from 'react';
 import { useStyles2 } from '@grafana/ui';
-import { useSelector } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import Page from 'app/core/components/Page/Page';
 import { usePerconaNavModel } from 'app/percona/shared/components/hooks/perconaNavModel';
-import { getPerconaSettings } from 'app/percona/shared/core/selectors';
+import { getPerconaServer, getPerconaSettings } from 'app/percona/shared/core/selectors';
 import { getSettingsStyles } from 'app/percona/settings/Settings.styles';
 import { FeatureLoader } from 'app/percona/shared/components/Elements/FeatureLoader';
 import { Connected } from './Connected/Connected';
 import { Connect } from './Connect/Connect';
+import { PlatformService } from './Platform.service';
+import appEvents from 'app/core/app_events';
+import { AppEvents } from '@grafana/data';
+import { logger } from '@percona/platform-core';
+import { fetchServerInfoAction, fetchSettingsAction, updateSettingsAction } from 'app/percona/shared/core/reducers';
+import { ConnectRenderProps } from './types';
+import { CONNECT_AFTER_SETTINGS_DELAY, CONNECT_DELAY } from './Platform.constants';
+import { Messages } from './Platform.messages';
 
 export const Platform: FC = () => {
   const navModel = usePerconaNavModel('settings-percona-platform');
   const settingsStyles = useStyles2(getSettingsStyles);
   const { result } = useSelector(getPerconaSettings);
+  const [connecting, setConnecting] = useState(false);
+  const { serverId: pmmServerId = '' } = useSelector(getPerconaServer);
+  const { result: settings, loading: settingsLoading } = useSelector(getPerconaSettings);
+  const dispatch = useDispatch();
+  const showMonitoringWarning = useMemo(() => settingsLoading || !settings?.publicAddress, [
```

Comment:
I noticed that you are using same logic in Connect.tsx. Isn't it worth to extract it? To custom hook for example including that useSelector?

Reply:
Yep, good point 👍 
Created a hook for this